recipes/libcurl/all/conanfile.py

```diff
@@ -225,35 +234,41 @@ def imports(self):
         #     but does not work on OS X 10.11 with SIP)
         # 2. copying dylib's to the build directory (fortunately works on OS X)
         if self.settings.os == "Macos":
-            self.copy("*.dylib*", dst=self._source_subfolder, keep_path=False)
+            copy(self, "*.dylib*", src=self.build_folder, dst=self.source_folder, keep_path=False)
```

Comment:
I'm not sure if `src=self.build_folder` is appropriate here...

Reply:
indeed, need to be checked ...

Reply:
Is it even needed ? I added the `fix_apple_shared_install_name()` call here: https://github.com/conan-io/conan-center-index/blob/a8c0f7eeff80fdf4bc5933d71bfaf4274b16e0ab/recipes/libcurl/all/conanfile.py#L585

After digging where the code comes from it is there since @jgsogo first import from bincrafters. Maybe we can remove it ?

Reply:
I wouldn't trust that guy 😄 

The easiest way to know is to test it (using shared OpenSSL). The problem here is the SIP protection in MacOS that doesn't propagate `*LD_LIBRARY_PATH` env-variables to other applications called from inside build scripts (CMake, autotools,...). This is not about `libcurl` libraries themselves but about dependencies whose `dylib` are not found unless something is added to those `...PATH`.

Reply:
> The problem here is the SIP protection in MacOS that doesn't propagate *LD_LIBRARY_PATH env-variables to other applications called from inside build scripts (CMake, autotools,...)

SIP was triggered by the way conan was calling CMake & autotools, with conan v2 and VirtualBuildEnv it should work fine.